### ISSUE TITLE
@prop default broken when field is a Map of Mixed

### ISSUE THREAD

**Issue by trappar** [typegoose/typegoose#308]
## Versions

- NodeJS: 14.3.0
- Typegoose(NPM): 7.2.0
- mongoose: 5.9.20
- mongodb: v4.0.3

## What is the Problem?

I'm trying to create a map in the shape of `Map<string, any>` in my schema by using `mongoose.Schema.Types.Mixed` as my `type`. In this configuration the "default" configuration is not being called at all. I've tested this with both the value and function variants of that configuration and nothing works as it usually does. Strangely, the configuration is completely valid in every other way.

I realize that this is not necessarily the typical use-case for maps, but it's valid config and should be supported.

## Code Example

```ts
import {getModelForClass, modelOptions, prop, Severity} from "@typegoose/typegoose";
import {Schema} from "mongoose";

@modelOptions({options: {allowMixed: Severity.ALLOW}})
class Test {
    @prop({
        type: Schema.Types.Mixed,
        default: () => {
            return new Map(); // This line is never called
        },
    })
    public things!: Map<string, any>;
}
const TestModel = getModelForClass(Test);

const doc = new TestModel();
console.log(doc.things); // undefined

// Works fine with manual initialization
doc.things = new Map();
doc.things.set('a', {something: 'arbitrary'});
console.log(doc.things); // Map(1) { 'a' => { something: 'arbitrary' } }
```

## Do you know *why* it happenes?

*no*

**Comment by trappar** [comment]
Though I don't know enough to understand the internals of this library at this point I've narrowed the issue down to this block of code: https://github.com/typegoose/typegoose/blob/5272903e5694eff393636de08f10b17a93101072/src/internal/utils.ts#L462-L470

I've found that this block isn't triggering because in this case:

```ts
OptionsCTOR.prototype instanceof (mongoose as any).SchemaTypeOptions // false
OptionsCTOR === (mongoose as any).SchemaTypeOptions // true
```

No options get applied at all, but even if the `if` yielded true, that block still doesn't work. It appears to be inappropriately handling the default value by assigning default to `ret.inner` rather than `ret.outer` where I'd want it to be. It looks like the logic there is that if the configuration can be applied to the `inner` options (which I'm guessing is the "of" type options), then it gets applied there. The problem is that there are some options like default which technically can apply to both.

So the following code generates my desired schema correctly, but it would undoubtedly break other schemas (assuming that the if works in some other situation):

```ts
if (OptionsCTOR === (mongoose as any).SchemaTypeOptions) {
    for (const [key, value] of Object.entries(options)) {
        ret.outer[key] = value;
    }
}
```

In conclusion, the issue is definitely that Typegoose is not creating the correct Mongoose schema from my prop configuration. I don't understand the constructor checking part, but regardless it seems like this may not be possible with the current structure since there is no way to specify that I want `default` to be applied to the map schema, not the "of" schema (if I'm understanding the code correctly).

Hope this helps with tracking down the issue.

**Comment by hasezoey** [comment]
the problem is that the `mongoose.Schema.Types.Mixed` type dosnt extend / use `SchemaOptions` - which is needed to map the options accordingly

but yes, there is an issue with it, when the type is not extending it, no options are mapped to anywhere

**Comment by hasezoey** [comment]
an "fix" will be included in 7.3.0
(if the types does not extend SchemaTypeOptions, default all options to outer-layer)

**Comment by trappar** [comment]
Thanks for the quick fix!
